grype/db/v6/internal/db.go

```diff
@@ -14,7 +14,7 @@ func NewDB(dbFilePath string, models []any, truncate bool) (*gorm.DB, error) {
 		return nil, err
 	}
 
-	if len(models) > 0 {
+	if len(models) > 0 && truncate {
```

Comment:
This line, and probably this method, could use a comment that we have only 2 modes of interacting with the DB which shouldn't mix:

1. Read mode for Grype at runtime should not migrate, vacuum, etc to avoid changing checksum or other corruption
2. Writing via grype-db should truncate the path to an empty db, apply all migration, migrate.